### ISSUE TITLE
Update `security.pam.enableSudoTouchIdAuth` for Sonoma+

### ISSUE THREAD

**Issue by citizen428** [LnL7/nix-darwin#1109]
Since macOS 14 (Sonoma), one can modify `/etc/pam.d/sudo_local` which will survive OS upgrades. By default, there's a file called `/etc/pam.d/sudo_local.template` with the following contents:

```
# sudo_local: local config file which survives system update and is included for sudo
# uncomment following line to enable Touch ID for sudo
#auth       sufficient     pam_tid.so
```

This gets included in `/etc/pam.d/sudoers`:

```
auth       include        sudo_local
```

It seems like a good idea to use this file so Touch ID for sudo will be available also right after upgrades, without having to run `darwin-rebuild switch` first.

**Comment by purcell** [comment]
I think this could be easily accomplished by making `file` [here](https://github.com/LnL7/nix-darwin/blob/a001f44cfc47164839eb61c6b1e7f4288813f7e8/modules/security/pam.nix#L20) conditional on `config.system.darwinVersion`.

**Comment by citizen428** [comment]
Yes, that should work, I briefly played around with that idea earlier today. In the end I just settled for this and removed the setting from my `flake.nix`, which is GoogEnough™ for me.

```nix
environment.etc = {
  "pam.d/sudo_local".text = ''
    auth sufficient pam_tid.so
  '';
};
```

**Comment by emilazy** [comment]
We should unconditionally generate `sudo_local` and retain only enough logic to add the `auth       include        sudo_local` line to `/etc/pam.d/sudo` when absent. See https://github.com/LnL7/nix-darwin/pull/1020 which I just realized I forgot to get around to reviewing again, oops :(

**Comment by citizen428** [comment]
Ugh, I only searched issues, not PRs, sorry. But I like your suggestion @emilazy, I think that's the best way forward.

**Comment by Enzime** [comment]
Duplicate of #784
